payments/payments-explained.md

```diff
@@ -0,0 +1,80 @@
+
+### Pay on agreement termination
+
+The simplest way to achieve payments for resources actually used within the scope of a single agreement is for the provider agent to wait until the agreement is terminated (for whatever reason) and issue a single invoice based on the registered usages and the agreed-upon coefficients that translate the usage into the final cost.
+
+The requestor agent then accepts such an invoice and pays the accumulated cost with a single transaction.
+
+And that is more or less what Golem's agents have been doing since we first released the new Golem, a.k.a yagna and its APIs.
+
+### Transaction costs
+
+One important factor influencing any decisions regarding the payment solutions in Golem is the fact that because all the payments are made using the GLM token, apart from just paying the providers, the requestor also has to bear the cost of the blockchain's transactions fees.
+
+Even though, by transitioning first to zkSync and more recently to Polygon, we were able to cut the transaction cost by orders of magnitude, they still constitute a substantial portion of the value of the payment itself.
+
+Naturally, the requestor's aim in such a scenario is to optimize this cost by performing as few transactions as possible. 
+
+Contrariwise, the providers are equally obviously interested in transactions happening as often as possible to ensure they are not left with uncovered, already incurred costs should the agreement be broken by the other side.
+
+### Keep-alive
+
+To partially address this problem, very early on, we have introduced a keep-alive mechanism where providers periodically issue debit notes to requestors in relatively short intervals. Those debit notes don't immediately trigger payments but they still carry the accumulated cost information.
+
+That way, on one end, the requestors have a potential to compare the reported cost with their own assumptions. On the other, by having those debit notes accepted, the providers can be sure that the requestors are still there and are willing to pay the final invoice.
+
+In this setup, if a requestor fails to accept a debit note before a set timeout, the provider may assume the requestor is no longer interested in keeping the agreement alive and thus, may terminate the agreement at will to stop accumulating the cost that now has little chance to be paid for.
+
+As you may see, this solution is still suboptimal. It still requires the requestor to be, one one hand, present and responsive to provider's debit notes and, on another, to be honest and pay the invoice that the provider issues at the very end. While this might be okay for agreements spanning from minutes to several hours, it definitely stops being okay when those agreeements may be signed for periods of days or even months. After all, it would be hard to expect the provider to continue working for a requestor based on a trust alone.
```

Comment:
```suggestion
As you may see, this solution is still suboptimal. It still requires the requestor to be, one one hand, present and responsive to provider's debit notes and, on another, to be honest and pay the invoice that the provider issues at the very end. While this might be okay for agreements spanning from minutes to several hours, it definitely stops being okay when those agreements may be signed for periods of days or even months. After all, it would be hard to expect the provider to continue working for a requestor based on a trust alone.
```